tests/testthat/test_model_conversions.R

```diff
@@ -0,0 +1,71 @@
+context("Unit Conversion Refactoring Tests")
+
+# These tests validate the refactoring to use PEcAn.utils::ud_convert()
+# for consistent unit conversions across SIPNET, DALEC, GDAY, and FATES models.
+
+test_that("UDUNITS2 recognizes new unit strings", {
+  library(PEcAn.utils)
+  
+  # Pool conversions (mass only)
+  expect_equal(ud_convert(100, "g/m2", "kg/m2"), 0.1)
+  expect_equal(ud_convert(1000, "g/m2", "kg/m2"), 1.0)
```

Comment:
I do not think we need to check the same unit conversion at multiple values -- Most of the actual conversion machinery is provided by the external `units` package and we trust them to test their own implementation, so our goal here should be to verify support for units that are important to us rather than to exhaustively test specific input values.